packages/notifications/src/components/NotificationMenu/index.tsx

```diff
@@ -1,86 +1,101 @@
-import React, { useRef, useState } from 'react';
+import React, { useImperativeHandle, useRef, useState } from 'react';
 import InternalOverlay from 'react-rainbow-components/components/InternalOverlay';
 import { BadgeOverlay, ButtonIcon, RenderIf } from 'react-rainbow-components';
 import { Bell } from '@rainbow-modules/icons';
 import { useOutsideClick } from '@rainbow-modules/hooks';
 import {
     ButtonIconHandle,
     NotificationMenuProps,
+    NotificationMenuHandles,
     Notification as NotificationInterface,
 } from './types';
 import { NotificationMenuContainer, StyledSpinner } from './styled';
 import Dropdown from './dropdown';
 import positionResolver from './helpers/positionResolver';
 
-const NotificationMenu: React.FC<NotificationMenuProps> = ({
-    label,
-    footer,
-    unreads,
-    isLoading,
-    notifications,
-    onClick,
-    className,
-    style,
-}: NotificationMenuProps) => {
-    const [isOpen, setIsOpen] = useState(false);
-    const buttonRef = useRef<ButtonIconHandle>();
-    const dropdownRef = useRef<HTMLElement>(null);
-    const inProgress = notifications?.some((value) => value.status === 'inProgress');
-    const icon = (
-        <BadgeOverlay overlap="circle" isHidden={!unreads}>
-            <Bell />
-        </BadgeOverlay>
-    );
+const NotificationMenu: React.ForwardRefExoticComponent<NotificationMenuProps> = React.forwardRef<
+    NotificationMenuHandles,
+    NotificationMenuProps
+>(
+    (
+        {
+            label,
+            footer,
+            unreads,
+            isLoading,
+            notifications,
+            onClick,
+            className,
+            style,
+        }: NotificationMenuProps,
+        ref,
```

Comment:
in order to make this more readable you can pass here props and ref, then make props destructuring inside the function body